addons/account/models/account_move.py

```diff
@@ -1728,7 +1728,8 @@ def create_tax_cash_basis_entry(self, percentage_before_rec):
         move_date = self.debit_move_id.date
         newly_created_move = self.env['account.move']
         with self.env.norecompute():
-            for move in (self.debit_move_id.move_id, self.credit_move_id.move_id):
+            # We use a set here in case the reconciled lines belong to the same move (it happens with POS)
+            for move in {self.debit_move_id.move_id, self.credit_move_id.move_id}:
```

Comment:
Detail: this would probably be more 'recordset-style' and have the same effect:

```suggestion
            for move in (self.debit_move_id.move_id | self.credit_move_id.move_id):
```